include/swift/IDE/Utils.h

```diff
@@ -293,6 +293,42 @@ class DeclNameViewer {
   unsigned partsCount() const { return 1 + Labels.size(); }
   unsigned commonPartsCount(DeclNameViewer &Other) const;
 };
+
+/// This provide a utility for writing to a underlying string buffer mulitiple
+/// string pieces and retrieve them later when the underlying buffer is stable.
+class DelayedStringRetriever : public raw_ostream {
+    SmallVectorImpl<char> &OS;
+    std::unique_ptr<llvm::raw_svector_ostream> Underlying;
+    SmallVector<std::pair<unsigned, unsigned>, 4> StartEnds;
+    unsigned Pos;
+    unsigned CurrentStart;
+public:
+    explicit DelayedStringRetriever(SmallVectorImpl<char> &OS) : OS(OS) {}
+    ~DelayedStringRetriever() override { assert(!Underlying); }
+    void startPiece() {
+      CurrentStart = OS.size();
+      Underlying.reset(new llvm::raw_svector_ostream(OS));
```

Comment:
`raw_svector_ostream` has flush() deleted. I think we don't need to call it here.